### PR TITLE
Trim reconciled records

### PR DESCRIPTION
**Pull request by mikew94**
Implemented service that will run as a cleanup for the Metadata Store.

The service is on a separate Spring profile allowing for us to toggle what is running through Terraform.

Service will delete data in Metadata Store that is older than a supplied scale (e.g. 1) and unit (e.g. week).

### CODE REVIEW

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/configuration/HBaseConfiguration.kt [hunk 1]
```diff
@@ -7,10 +7,12 @@ import org.apache.hadoop.hbase.client.ConnectionFactory
 import org.springframework.boot.context.properties.ConfigurationProperties
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
 import uk.gov.dwp.dataworks.logging.DataworksLogger
 
 @Configuration
 @ConfigurationProperties(prefix = "hbase")
+@Profile("HBASE")
```
Comment (DanielChicot) [inline src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/configuration/HBaseConfiguration.kt:15]: Is there an associated infra change in place to activate this?

File: src/test/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImplTest.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @MockBean
+    private lateinit var metadataStoreRepository: MetadataStoreRepository
+
+    @Test
+    fun willDeleteRecordsOlderThanScaleAndUnit() {
+
+        whenever(metadataStoreRepository.deleteRecordsOlderThanPeriod()).thenReturn(1)
+
+        service.trimReconciledRecords()
+
```
Comment (DanielChicot) [inline src/test/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImplTest.kt:29]: Can we assert that the metadataStoreRepository methods are invoked with the appropriate methods?

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/MetadataStoreRepository.kt [hunk 3]
```diff
@@ -9,8 +10,13 @@ import java.sql.ResultSet
 import java.sql.Timestamp
 
 @Repository
-class MetadataStoreRepository(private val connection: Connection,
-                              private val table: String) {
+class MetadataStoreRepository(
+    private val connection: Connection,
+    @Qualifier("table") private val table: String,
+    @Qualifier("queryLimit") private val queryLimit: String,
+    @Qualifier("trimReconciledScale") private val trimReconciledScale: String,
```
Comment (DanielChicot) [inline src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/MetadataStoreRepository.kt]: Can these 2 trim variables be passed to the TrimService which in turn passes them to the metadatastore deleteRecords method?

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImpl.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import uk.gov.dwp.dataworks.kafkatohbase.reconciliation.repositories.MetadataStoreRepository
+import uk.gov.dwp.dataworks.kafkatohbase.reconciliation.services.ReconciliationService
+import uk.gov.dwp.dataworks.kafkatohbase.reconciliation.services.TrimReconciledRecordsService
+import uk.gov.dwp.dataworks.logging.DataworksLogger
+
+@Service
+@Profile("TRIM_RECONCILED_RECORDS")
+class TrimReconciledRecordsServiceImpl(
+    private val reconcilerConfiguration: ReconcilerConfiguration,
+    private val metadataStoreRepository: MetadataStoreRepository
```
Comment (DanielChicot) [inline src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImpl.kt]: Can we have the trim string variables passed here and not into the metadatastore

**Review by DanielChicot — CHANGES_REQUESTED** [review]

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/configuration/HBaseConfiguration.kt (hunk 1 above)
Comment (mikew94) [reply src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/configuration/HBaseConfiguration.kt:15]: Yeh I have that on the side, will post that shortly before this gets merged

File: src/test/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImplTest.kt (hunk 2 above)
Comment (mikew94) [reply src/test/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImplTest.kt:29]: Yeh will do

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/MetadataStoreRepository.kt (hunk 3 above)
Comment (mikew94) [reply src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/MetadataStoreRepository.kt]: Yeh can do that

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImpl.kt (hunk 4 above)
Comment (mikew94) [reply src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/services/impl/TrimReconciledRecordsServiceImpl.kt]: Indeedio

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/configuration/HBaseConfiguration.kt (hunk 1 above)
Comment (DanielChicot) [reply src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/configuration/HBaseConfiguration.kt:15]: Cool

File: src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/MetadataStoreRepository.kt (hunk 3 above)
Comment (DanielChicot) [reply src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/MetadataStoreRepository.kt]: Thanks.

**Review by DanielChicot — APPROVED** [review]

**Pull request merged**